### PR TITLE
Fix addcmul/addcdiv/addmm opinfos to not pass float scalar when an integer dtype is requested.

### PR DESCRIPTION
**Pull request by Chillee**
Fixes https://github.com/pytorch/pytorch/issues/72915

### CODE REVIEW

**Comment by ngimel** [comment]
Can you also make addcmul/addcdiv throw error if `value` is of the wrong type, like `add/sub` does?

**Comment by Chillee** [comment]
@ngimel That would require C++ changes, right?

**Comment by ngimel** [comment]
yes

**Comment by Chillee** [comment]
@ngimel Any pointers to what I should modify?

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -2077,8 +2077,11 @@ def make_arg_conj(size):
 
 
 def sample_inputs_addmm(op_info, device, dtype, requires_grad, **kwargs):
-    alpha_val = kwargs.get('alpha', 2 + 3j if dtype.is_complex else 0.6)
-    beta_val = kwargs.get('beta', 1 + 2j if dtype.is_complex else 0.2)
+    alpha_val = kwargs.get('alpha', 2 + 3j if dtype.is_complex else 2.6)
+    beta_val = kwargs.get('beta', 1 + 2j if dtype.is_complex else 3.2)
+    if not dtype.is_floating_point:
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:2082]: These checks should be if the datatype is neither floating point nor complex

**Comment by mruberry** [comment]
If this PR updates the error handling of these operations it's a good opportunity to add ErrorInputs for the disallowed case(s). See 

https://github.com/pytorch/pytorch/blob/bbac8c9c4816c87eae17a04b60d4103940c08f83/torch/testing/_internal/common_methods_invocations.py#L2983

for an example. Although that's a relatively complicated case.

**Comment by ngimel** [comment]
Add `alpha_check` in ATen/native/PointwiseOps.cpp, similar to https://github.com/pytorch/pytorch/blob/352eeb2ef92f2d98d15792d15e6f8067945d6b85/aten/src/ATen/native/BinaryOps.cpp#L48

**Comment by Chillee** [comment]
@ngimel Unfortunately, doesn't seem trivial to do so, since `alpha_check` is in `BinaryPointwiseOps.h`, which is not imported by `LinearAlgebra.cpp` or `PointwiseOps.cpp`. And I don't see an obvious candidate to move `alpha_check` so that all 3 files can import it.

I could just copy paste it in 3 places, but that seems even worse.

So I'd prefer to not do it in this PR.

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -2077,8 +2077,20 @@ def make_arg_conj(size):
...
 def sample_inputs_addmm(op_info, device, dtype, requires_grad, **kwargs):
-    alpha_val = kwargs.get('alpha', 2 + 3j if dtype.is_complex else 0.6)
-    beta_val = kwargs.get('beta', 1 + 2j if dtype.is_complex else 0.2)
+    if dtype.is_complex:
+        alpha_val = 2 + 3j
+        beta_val = 1 + 2j
+    elif dtype.is_floating_point:
+        alpha_val = 2.6
+        beta_val = 3.2
+    elif not dtype.is_floating_point:
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:2086]: This conditional is a little weird -- `if elif else` might be more natural? Or writing with more readable variable scoping

```
alpha_val = 2
beta_val = 3
if dtype.is_complex:
  ...
elif dtype.is_floating_point:
  ...
```

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -2218,7 +2230,7 @@ def sample_inputs_addcmul_addcdiv(op_info, device, dtype, requires_grad, **kwarg
         sample_inputs.append(SampleInput(
             args[0],
             args=args[1:],
-            kwargs=dict(value=3.14), broadcasts_input=broadcasts_input))
+            kwargs=dict(value=3.14 if dtype.is_floating_point else 3), broadcasts_input=broadcasts_input))
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:2233]: floating point or complex?

**Comment by ngimel** [comment]
> @ngimel Unfortunately, doesn't seem trivial to do so, since `alpha_check` is in `BinaryPointwiseOps.h`, which is not imported by `LinearAlgebra.cpp` or `PointwiseOps.cpp`. And I don't see an obvious candidate to move `alpha_check` so that all 3 files can import it.

Just put it in a small header in aten/src/ATen, it already has a few, and import that header at all call sites.

**Comment by ezyang** [comment]
what's the status on this?

**Comment by Chillee** [comment]
Will finish and respond to comments. Have fast compiles now so happy to modify C++ now :^)

**Pull request closed without merging**